node/pkg/admin/aggregator/controller.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/go-playground/validator"
+	"github.com/gofiber/fiber/v2"
+)
+
+type AggregatorModel struct {
+	Id     *int64 `db:"id" json:"id"`
+	Name   string `db:"name" json:"name"`
+	Active bool   `db:"active" json:"active"`
+}
+
+type AggregatorInsertModel struct {
+	Name string `db:"name" json:"name" validate:"required"`
+}
+
+func insert(c *fiber.Ctx) error {
+	payload := new(AggregatorInsertModel)
+	if err := c.BodyParser(payload); err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to parse body for aggregator insert payload: " + err.Error())
+	}
+
+	validate := validator.New()
+	if err := validate.Struct(payload); err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to validate aggregator insert payload: " + err.Error())
+	}
+
+	result, err := db.QueryRow[AggregatorModel](c.Context(), InsertAggregator, map[string]any{
+		"name": payload.Name,
+	})
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator insert query: " + err.Error())
+	}
+
+	return c.JSON(result)
+}
+
+func get(c *fiber.Ctx) error {
+	result, err := db.QueryRows[AggregatorModel](c.Context(), GetAggregator, nil)
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator get query: " + err.Error())
+	}
+
+	return c.JSON(result)
+}
+
+func getById(c *fiber.Ctx) error {
+	id := c.Params("id")
+	result, err := db.QueryRow[AggregatorModel](c.Context(), GetAggregatorById, map[string]any{"id": id})
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator get by id query: " + err.Error())
+	}
+	return c.JSON(result)
+}
+
+func deleteById(c *fiber.Ctx) error {
+	id := c.Params("id")
+	result, err := db.QueryRow[AggregatorModel](c.Context(), DeleteAggregatorById, map[string]any{"id": id})
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator delete by id query: " + err.Error())
+	}
+	return c.JSON(result)
+}
+
+func syncWithAdapter(c *fiber.Ctx) error {
+	result, err := db.QueryRows[AggregatorModel](c.Context(), SyncAggregator, nil)
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator sync with adapter query: " + err.Error())
+	}
+	return c.JSON(result)
+}
+
+func activate(c *fiber.Ctx) error {
+	id := c.Params("id")
+	result, err := db.QueryRow[AggregatorModel](c.Context(), ActivateAggregator, map[string]any{"id": id})
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to execute aggregator activate query: " + err.Error())
+	}
+
+	err = utils.SendMessage(c, bus.AGGREGATOR, bus.ACTIVATE_AGGREGATOR, map[string]any{"id": id})
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).SendString("failed to send message to aggregator: " + err.Error())
+	}
+
+	time.Sleep(10 * time.Millisecond)
```

Comment:
@nick-bisonai Can you please explain why do we need to sleep here?

Reply:
From adapter test code, when I executed bus msg publish without any delay through for loop, it lead to an error. from receiver perspective, when published 2 messages through for loop, it received same messages twice. but after adding slight delay it was solved

Reply:
Thank you for explanation! Let's open an issue for this because it probably just hides some bigger problem 🙏